waspc/src/Wasp/Analyzer/Parser/Valid.hs

```diff
@@ -0,0 +1,27 @@
+module Wasp.Analyzer.Parser.Valid
+  ( validateAst,
+  )
+where
+
+import Data.List (find)
+import qualified Wasp.Analyzer.Parser as P
+
+validateAst :: P.AST -> Either (String, P.Ctx) P.AST
+validateAst = validateNoEntityDeclInWaspFile
+
+validateNoEntityDeclInWaspFile :: P.AST -> Either (String, P.Ctx) P.AST
+validateNoEntityDeclInWaspFile ast@(P.AST stmts) = case findEntityDecl stmts of
+  Just (P.WithCtx ctx _) -> Left (entitiesNoLongerSupportedError, ctx)
+  Nothing -> Right ast
+  where
+    findEntityDecl :: [P.WithCtx P.Stmt] -> Maybe (P.WithCtx P.Stmt)
+    findEntityDecl = find isEntityDecl
+
+    isEntityDecl :: P.WithCtx P.Stmt -> Bool
+    isEntityDecl = \case
+      P.WithCtx _ (P.Decl "entity" _ _) -> True
```

Comment:
How come we need to hardcode "entity" here? I haven't looked at this code for a long time so it might be necessary, but it seems weird.

Reply:
Well spotted! We should make sure that the entity declaration type name (`"entity"`) is the same in the:
- definition of the entity declaration type
- here.

I'll make it happen 👍